ocular_cli/src/commands/chains/add.rs

```diff
@@ -1 +1,72 @@
+use super::show::ShowCmd;
+use crate::{config, prelude::*};
+use abscissa_core::{Command, Runnable};
+use clap::Parser;
+use ocular::chain::{info::ChainInfo, registry};
+use serde::{Deserialize, Serialize};
+use std::{fs, io::Write, path::Path};
 
+#[derive(Command, Debug, Parser)]
+pub struct AddCmd {
+    name: String,
+}
+
+#[derive(Deserialize, Serialize)]
+struct Chains {
+    chains: Vec<ChainInfo>,
+}
+
+impl Runnable for AddCmd {
+    /// Add chain to local config file
+    fn run(&self) {
+        // Navigate to file of local config
+        let path = config::get_config_path();
+        let config_file = Path::new(path.to_str().unwrap());
+
+        // Check if chain already exists
+        let chain_content = fs::read_to_string(config_file).unwrap_or_else(|err| {
+            status_err!("Can't read config file: {}", err);
+            std::process::exit(1);
+        });
+
+        let chain_name = self.name.as_str();
+
+        abscissa_tokio::run(&APP, async {
+            let chain_info = registry::get_chain(chain_name).await.unwrap_or_else(|err| {
+                status_err!("Can't fetch chain from chain registry: {}", err);
+                std::process::exit(1);
+            });
+
+            let mut vec = Vec::new();
+            vec.push(chain_info);
+
+            if !chain_content.contains(chain_name) {
```

Comment:
If the chain exists in the config we can avoid a call to github by putting this before the async block.